org.mwc.cmap.NarrativeViewer/src/org/mwc/cmap/NarrativeViewer/app/NViewerView.java

```diff
@@ -967,11 +974,12 @@ protected void timeUpdated(final HiResDate dtg)
 
             public void run()
             {
-              if (_pendingTime != null)
+              if (_pendingTime.get() != INVALID_TIME)
               {
                 // quick, capture the time
-                final Long safeTime = _pendingTime;
-                _pendingTime = null;
+                final Long safeTime = _pendingTime.get();

```

Comment:
@IanMayo move final Long safeTime = _pendingTime.get(); outside if and use it also in if condition